lib/src/barback/ddc_transformer.dart

```diff
@@ -0,0 +1,500 @@
+      return;
+    } else {
+      logger.info('Took ${watch.elapsed} to compile package:$basePackage '
+          'with dartdevc.');
+    }
+
+    watch.reset();
+    addOutput(
+        new Asset.fromString(jsOutputId, await jsOutputFile.readAsString()));
+    addOutput(new Asset.fromString(
+        sourceMapOutputId, await sourceMapOutputFile.readAsString()));
+    addOutput(new Asset.fromBytes(
+        summaryOutputId, await summaryOutputFile.readAsBytes()));
+    logger.fine(
+        'Took ${watch.elapsed} to copy output files for package:$basePackage.');
+  } catch (e) {
+    logger.error('$e');
+  } finally {
+    await tmpDir.delete(recursive: true);
+  }
+}
+
+void _createAmdBootstrap(AssetId entryPointId, AssetId bootstrapId,
+    AssetId moduleId, _OutputWriter addOutput) {
+  var appModuleName = p.withoutExtension(
+      p.relative(moduleId.path, from: p.dirname(entryPointId.path)));
+
+  // TODO(jakemac53): Sane module name creation, this only works in the most
+  // basic of cases.
+  //
+  // See https://github.com/dart-lang/sdk/issues/27262 for the root issue which
+  // will allow us to not rely on the naming schemes that dartdevc uses
+  // internally, but instead specify our own.
+  var appModuleScope = p.url
+      .split(moduleId.path.substring(0, moduleId.path.indexOf('.dart')))
+      .join("__");
+  var bootstrapContent = '''
+require(["$appModuleName", "dart_sdk"], function(app, dart_sdk) {
+  dart_sdk._isolate_helper.startRootIsolate(() => {}, []);
+  app.$appModuleScope.main();
+});
+''';
+  addOutput(new Asset.fromString(bootstrapId, bootstrapContent));
+
+  var bootstrapModuleName = p.withoutExtension(
+      p.relative(bootstrapId.path, from: p.dirname(entryPointId.path)));
+  var entryPointContent = '''
+var el = document.createElement("script");
+el.defer = true;
+el.async = false;
+el.src = "require.js";
+el.setAttribute("data-main", "$bootstrapModuleName");
+document.head.appendChild(el);
+''';
+  addOutput(new Asset.fromString(entryPointId, entryPointContent));
+}
+
+/// Copies [ids] to [tmpDir], and returns the set of [File]s that were created.
+Future<Set<File>> _createTmpFiles(
+    Set<AssetId> ids,
+    Directory tmpDir,
+    Directory packagesDir,
+    String basePackage,
+    _InputChecker hasInput,
+    _InputReader readInput,
+    TransformLogger logger) async {
+  var files = new Set<File>();
+  await Future.wait(ids.map((id) async {
+    if (!await hasInput(id)) {
+      logger.warning('Unable to find asset `$id` when compiling '
+          'package:$basePackage.');
+      return;
+    }
+    var file = _fileForId(id, tmpDir.path, packagesDir.path);
+    await _writeFile(file, readInput(id));
+    files.add(file);
+  }));
+  return files;
+}
+
+/// Crawls from [entryId] and finds all [Asset]s that are relative through
+/// relative uris (via all [UriBasedDirective]s).
+Future<Set<AssetId>> _findRelativeIds(AssetId entryId, TransformLogger logger,
```

Comment:
I don't see this, it looks like it just has a `CompilerInputProvider` for dart2js which can read assets by uri, but it doesn't actually follow the import graph and only handles absolute uris.